### PR TITLE
*: Walking code fix

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve? 

This is some trivial code change when I walking the code, such as fix typo, add comments or extract functions to try to make the code easier to read and reason.

### What is changed and how it works?

* fix some typo in code and comment
* extract some method from long line functions.

### Check List 

Tests 

 - No code

Code changes

Side effects

Related changes

### Release note 

```release-note
`None`
```

### CODE REVIEW

**Comment by ben1009** [comment]
IMHO ` Files changed 34` make the reviewer a little nervous to review the change

File: cdc/processor/manager.go [hunk 1]
```diff
@@ -33,14 +33,14 @@ import (
 type commandTp int
 
 const (
-	commandTpUnknow commandTp = iota //nolint:varcheck,deadcode
+	commandTpUnknown commandTp = iota //nolint:varcheck,deadcode
```
Comment (asddongmen) [inline cdc/processor/manager.go:36]: I think the abbreviation Tp here is confusing too.

File: cdc/owner/gc_manager.go [hunk 2]
```diff
@@ -67,29 +67,34 @@ func newGCManager() *gcManager {
...
 }
 
-func (m *gcManager) updateGCSafePoint(ctx cdcContext.Context, state *model.GlobalReactorState) error {
-	if time.Since(m.lastUpdatedTime) < gcSafepointUpdateInterval {
-		return nil
-	}
+func getMinCheckpointTs(feeds map[model.ChangeFeedID]*model.ChangefeedReactorState) uint64 {
 	minCheckpointTs := uint64(math.MaxUint64)
-	for _, cfState := range state.Changefeeds {
+	for _, cfState := range feeds {
```
Comment (asddongmen) [inline cdc/owner/gc_manager.go:72]: may be changefeeds is better.

File: cdc/owner/changefeed.go [hunk 3]
```diff
@@ -120,13 +120,10 @@ func (c *changefeed) Tick(ctx cdcContext.Context, state *model.ChangefeedReactor
 }
 
 func (c *changefeed) checkStaleCheckpointTs(ctx cdcContext.Context, checkpointTs uint64) error {
-	state := c.state.Info.State
-	if state == model.StateNormal || state == model.StateStopped || state == model.StateError {
-		if err := c.gcManager.checkStaleCheckpointTs(ctx, checkpointTs); err != nil {
-			return errors.Trace(err)
-		}
+	if c.state.Info.State.Runnable() {
+		return nil
```
Comment (asddongmen) [inline cdc/owner/changefeed.go:124]:
```suggestion
	if !c.state.Info.State.Runnable() {
		return nil
```

File: cdc/owner/gc_manager.go [hunk 4]
```diff
@@ -67,29 +67,34 @@ func newGCManager() *gcManager {
...
 		if cfState.Info == nil {
```
Comment (asddongmen) [inline cdc/owner/gc_manager.go:73]:
```suggestion
		if cfState.Info == nil || !cfState.Info.State.Runnable() {
```

File: cdc/owner/gc_manager.go [hunk 5]
```diff
@@ -67,29 +67,34 @@ func newGCManager() *gcManager {
...
 			continue
 		}
-		switch cfState.Info.State {
-		case model.StateNormal, model.StateStopped, model.StateError:
-		default:
+		if !cfState.Info.State.Runnable() {
 			continue
 		}
+
```
Comment (asddongmen) [inline cdc/owner/gc_manager.go:79]:
```suggestion
```

File: cdc/owner/owner.go [hunk 6]
```diff
@@ -133,6 +133,7 @@ func (o *Owner) Tick(stdCtx context.Context, rawState orchestrator.ReactorState)
 		}
 		cfReactor.Tick(ctx, changefeedState, state.Captures)
 	}
+	// changefeed metadata already erased from etcd, so we have to delete those changefeed from memory.
```
Comment (asddongmen) [inline cdc/owner/owner.go:136]: may be we can extract a func to do this job.

File: pkg/txnutil/gc/gc_manager.go [hunk 7]
```diff
@@ -93,6 +93,8 @@ func (m *gcManager) TryUpdateGCSafePoint(
 		}
 		return nil
 	}
+	m.lastUpdatedTime = time.Now()
```
Comment (overvenus) [inline pkg/txnutil/gc/gc_manager.go:96]: Is it a bug fix?

Comment (amyangfei) [inline pkg/txnutil/gc/gc_manager.go:96]: Should L83 `m.lastUpdatedTime = time.Now()` be removed,
Is this a bug in some scenarios?

File: cdc/sink/txns_heap.go [hunk 8]
```diff
@@ -75,6 +82,7 @@ func (h *txnsHeap) iter(fn func(txn *model.SingleTableTxn)) {
 		bucket := minEntry.bucket
 		fn(h.txnsGroup[bucket][0])
 		h.txnsGroup[bucket] = h.txnsGroup[bucket][1:]
+		// TODO (Ling Jin): why we need this
```
Comment (overvenus) [inline cdc/sink/txns_heap.go:85]:
```suggestion
```

Remove or explain, do not leave a TODO question.

File: cdc/puller/metrics.go [hunk 9]
```diff
@@ -53,6 +54,7 @@ var (
 			Help:      "Puller entry buffer size",
 			Buckets:   prometheus.ExponentialBuckets(1, 2, 8),
 		}, []string{"capture", "changefeed"})
+	// todo (Ling Jin): memBufferSize is not used, add it to monitor
```
Comment (overvenus) [inline cdc/puller/metrics.go:57]:
```suggestion
	//TODO(Ling Jin): memBufferSize is not used, add it to monitor
```

Could you file an issue?

File: cdc/entry/mounter.go [hunk 10]
```diff
@@ -272,6 +272,7 @@ const (
 )
 
 // UnmarshalDDL unmarshals the ddl job from RawKVEntry
+// todo (Ling Jin): how to determine it's a valid ddl event
```
Comment (overvenus) [inline cdc/entry/mounter.go:275]:
```suggestion
// TODO(Ling Jin): how to determine it's a valid ddl event
```

What do you mean by "valid"?

File: cdc/processor/manager.go (hunk 1 above)
Comment (3AceShowHand) [reply cdc/processor/manager.go:36]: me too, let's change it to `commandType`

**Pull request closed without merging**